### ISSUE TITLE
Improve __repr__ of TypeVar

### ISSUE THREAD

**Issue by 3a73398b-c396-4cae-b465-208710eb5200** [python/cpython#91224]
BPO | [47068](https://bugs.python.org/issue47068)
--- | :---
Nosy | @ktbarrett

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2022-03-19.17:19:22.518>
labels = ['type-feature', '3.11']
title = 'Improve __repr__ of TypeVar'
updated_at = <Date 2022-03-19.17:19:22.518>
user = 'https://github.com/ktbarrett'
```

bugs.python.org fields:
```python
activity = <Date 2022-03-19.17:19:22.518>
actor = 'ktbarrett'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = []
creation = <Date 2022-03-19.17:19:22.518>
creator = 'ktbarrett'
dependencies = []
files = []
hgrepos = []
issue_num = 47068
keywords = []
message_count = 1.0
messages = ['415562']
nosy_count = 1.0
nosy_names = ['ktbarrett']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue47068'
versions = ['Python 3.11']
```

</p></details>

**Comment by sobolevn** [comment]
I think that it might be quite verbose, especially with multiple constraints. Let's say that you have 3 constaints with custom types:

```python
# my_module.py
class First: pass
class Second: pass
class Third: pass

_T = TypeVar('_T', First, Second, Third)
repr(_T)

# ~_T<=(my_module.First, my_module.Second, my_module.Third)
```

However, I like the idea. And I think that this should be made in line with new typing syntx propsed in typing-sig.

Noisy @JelleZijlstra

**Comment by Gobot1234** [comment]
Would now be an appropriate time to implement this?

**Comment by JelleZijlstra** [comment]
No, because we likely won't use the Python implementation of TypeVar anyway in 3.12: erictraut/cpython#3.

**Comment by Gobot1234** [comment]
After PEP 695 is merged into main then?

**Comment by JelleZijlstra** [comment]
I might actually prefer the current repr(), though I'd be open to an alternative if someone proposes something good enough.

With PEP 695 variance is mostly irrelevant: PEP 695 TypeVars have inferred variance and we don't insert any punctuation, so their repr() is just the name.

Including the bound/constraints in the repr could still be useful. The most obvious way to indicate the bound is as e.g. `T: int`, `T: (bytes, str)`. However, that doesn't look great in a compound repr, like `list[T: int]` (which looks like a slice). ktbarrett above suggested `<=` instead, which would solve the ambiguity but be inconsistent with PEP 695's native syntax for bounds.
